test/e2e/monitoring.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	influxdb "github.com/influxdb/influxdb/client"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("Monitoring", func() {
+	var c *client.Client
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+	})
+
+	It("verify monitoring pods and all cluster nodes are available on influxdb using heapster.", func() {
+		if testContext.provider != "gce" {
+			By(fmt.Sprintf("Skipping Monitoring test, which is only supported for provider gce (not %s)",
+				testContext.provider))
+			return
+		}
+		testMonitoringUsingHeapsterInfluxdb(c)
+	})
+})
+
+const (
+	influxdbService      = "monitoring-influxdb"
+	influxdbDatabaseName = "k8s"
+	influxdbUser         = "root"
+	influxdbPW           = "root"
+	podlistQuery         = "select distinct(pod) from stats"
+	nodelistQuery        = "select distinct(hostname) from machine"
+	sleepBetweenAttempts = 30 * time.Second
```

Comment:
I reduced the timeout to 5 minutes and sleep duration to 10s. If this test is run as soon as the cluster is setup, it can take some time for the pods to be active.